### PR TITLE
MueLu: Fix fscanf and system warnings

### PR DESCRIPTION
**Pull request by GrahamBenHarper**
@trilinos/muelu

I saw some warnings about `fscanf` unused returns and `system` unused returns originating from `MatrixLoad.hpp` and `AggExport.cpp` like the one below. I fixed them :)
```
/home/graham/Programming/cpp/Trilinos/Trilinos-source/packages/muelu/test/scaling/MatrixLoad.hpp:295:19: warning: ignoring return value of ‘int fscanf(FILE*, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]
  295 |             fscanf(fp,"%d %d %d\n",&nBlks, &nRows, &nnzs);
      |             ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Note: While I placed asserts at the start of the file parsing in `MatrixLoad.hpp`, I did not continue by placing asserts inside the main loops. In principle, they should be inside the loop, but I would rather not sacrifice performance on loading user input, although I have no clue how much of an impact it would have on performance anyway. Feel free to discuss if you think it's worth adding an assert in the main loop as well.

### CODE REVIEW

File: packages/muelu/test/scaling/MatrixLoad.hpp [hunk 1]
```diff
@@ -296,6 +296,7 @@ void readUserBlks(const std::string& userBlkFileName, const std::string& smoothe
           if (mueluList.sublist(smootherOrCoarse + ": params").sublist("subdomain solver parameters").get<std::string>("partitioner: type") == "user") {
 
             FILE   *fp;
+	    int    retVal;
```
Comment (cgcgcg) [inline packages/muelu/test/scaling/MatrixLoad.hpp:299]: not sure if github is playing tricks on me here: indentation?

Comment (GrahamBenHarper) [reply packages/muelu/test/scaling/MatrixLoad.hpp:299]: my emacs config might be messed up on this machine. how do I keep the default emacs from inserting tabs?

Comment (cgcgcg) [reply packages/muelu/test/scaling/MatrixLoad.hpp:299]: I have `(setq-default indent-tabs-mode nil)` in my config.

Comment (GrahamBenHarper) [reply packages/muelu/test/scaling/MatrixLoad.hpp]: okay it should be fixed now. thanks for that :)

**Comment by GrahamBenHarper** [comment]
My first PR from 38000ft. Let's see if it testing passes while I'm still airborne

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**